### PR TITLE
Add button components

### PR DESCRIPTION
**Pull request by ronilaukkarinen**
Button is now a mixin with variables like `$size` (small, medium, large and default) and `$ghost` (true/false). Example added to demo:

Screen-Shot-2021-02-16-11-52-01 76

This PR also fixes the consistencies in button aesthetics (pixel perfect paddings).

### CODE REVIEW

File: sass/components/_button.scss [hunk 1]
```diff
@@ -0,0 +1,103 @@
+@mixin button($color: $color-white, $background: $color-darkgrey, $radius: 3px, $border-width: 2px, $font-weight: 700, $size: default, $ghost: false) {
```
[the remark this answers is not in the document]
Comment (ronilaukkarinen) [reply sass/components/_button.scss:1]: That way you cannot use multiple styles with the same mixin?

File: sass/components/_button.scss [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    display: none;
+  }
+  /* stylelint-enable */
+
+  // Arrow icon
+  svg {
+    margin-left: 2rem;
+  }
+
+  @if $size == 'small' {
```
[the remark this answers is not in the document]
Comment (ronilaukkarinen) [reply sass/components/_button.scss:31]: Also thought about that, wasn't sure which way to go here but didn't think it through yet. Your approach with sizes and styles looks good, I vote for that!

File: sass/components/_button.scss [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    font-size: var(--font-size-small);
+    padding-bottom: 8px;
+    padding-left: 17px;
+    padding-right: 17px;
+    padding-top: 8px;
+
+    &:hover,
+    &:focus {
+      background-color: $color-buttons-hover;
```
[the remark this answers is not in the document]
Comment (ronilaukkarinen) [reply sass/components/_button.scss:40]: For sizes it doesn't make sense, you are correct. I may have mistakenly copied these over from styles. Good catch.

File: sass/components/_button.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (Niq1982) [reply sass/components/_button.scss:1]: Ah, true that. If we approach this the way I suggested on the other thread, using multiple styles won't be an issue. Also, the developer can then decide in which order the styles are being applied.

```
.foo-button {
  @include button();
  @include button-size-small();
  @include button-style-ghost();
}
```

File: sass/components/_button.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (ronilaukkarinen) [reply sass/components/_button.scss:31]: Took it even further and called the button mixin inside other mixin variants so now you only need to call variant for example style like this `@include button-size-small($radius: 50%);`, no need to default the actual button first. In a sense the original button is a dependency of size and style variant mixins. Beautiful.

[the remark this answers is not in the document]
Comment (ronilaukkarinen) [reply sass/components/_button.scss:31]: I take it back. With that previously mentioned approach you'd have to define each size and style separately if you'd want for example unified border-radiuses for all. Your approach is better. Let's not overcomplicate this.

**Pull request merged**